helm/templates/minio-external-scaler-deployment.yaml

```diff
@@ -0,0 +1,32 @@
+{{ if .Values.minio.externalscaler.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: minio-external-scaler
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: minio-external-scaler
+  template:
+    metadata:
+      labels:
+        app: minio-external-scaler
+    spec:
+      containers:
+      - name: minio-external-scaler
+        image: {{ $.Values.minio.externalscaler.image }}
+        imagePullPolicy: {{ $.Values.minio.externalscaler.imagePullPolicy }}
+        resources: {{ toYaml $.Values.minio.externalscaler.resources | nindent 10 }}
+        env:
+        - name: CWM_REDIS_HOST
+          value: {{ $.Values.minio.externalscaler.REDIS_HOST }}
```

Comment:
this should be from the metricsLogger.REDIS_HOST

it has to be the same redis as the metricsLogger

Reply:
The Redis is deployed along with the logger in the same pod so its host address `localhost` works for the logger.
But, the external scaler is deployed separately.
How would `metricsLogger.REDIS_HOST` work for the scaler in this case?

Reply:
it's localhost only for development / testing when .Values.minio.metricsLogger.withRedis is true, in production it uses a different REDIS_HOST

you should change it's default value instead of localhost to be `minio-logger` that way it will work for both

Reply:
With `REDIS_HOST: minio-logger`, the minio's audit webhook is failing and the logger's fluentd instance is unable to locate the Redis server to connect with. Also, restarting `coredns` service doesn't help.

Here are the steps to reproduce this:

1. `minikube start --driver=docker --kubernetes-version=v1.16.14`
2. `eval $(minikube -p minikube docker-env)`
3. `export COMPOSE_TLS_VERSION=TLSv1_2`
4. `docker-compose build`
5. Configure and deploy with the scaler first i.e. `externalscaler.enabled=true`:

```yaml
minio:
  image: minio
  tag: latest
  initDebugEnable: true
  metricsLogger:
    image: cwm-worker-logger
    tag: latest
    LOG_LEVEL: debug
    LOG_PROVIDER: default
    UPDATE_GRACE_PERIOD_SECONDS: 2s
    DEPLOYMENT_API_METRICS_FLUSH_INTERVAL_SECONDS: 5s

  externalscaler:
    enabled: true

  scaledobject:
    enabled: false
```

6. Deploy: `helm upgrade -f .values.yaml --install cwm-worker-deployment-minio ./helm`
7. The above step should reproduce the issue. The same errors can be seen in the [CI workflow](https://github.com/CloudWebManage/cwm-worker-deployment-minio/pull/17/checks?check_run_id=2575852170#step:3:673). The minio pod keeps failing at audit webhook. And, the logs from the logger pod will show that it keeps trying to connect with the Redis server.
8. If that works, the next will be to enable ScaledObject i.e. `scaledobject.enabled=true` and deploy it again.

Reply:
I guess Kubernetes doesn't have a route from pod to it's own service.. I modified it to use localhost hard-coded if withRedis is true.. (withRedis is only used for testing so there is no need to specify a different REDIS_HOST in this case)

I also did some other small changes:

* updated README with instructions to build the external scaler image
* force a restart of logger deployment in case of change to logger configmap

Reply:
Right. Just viewed the changes. Looks good.